scanner.js

```diff
@@ -185,15 +190,26 @@ function scan(line, linenumber, tokens) {
       emit(line[pos++])
     }
 
-    // Boolean literals
-    else if (booleanLit.test(trueTest) || booleanLit.test(falseTest)) {
-      if (booleanLit.test(trueTest)) {
-        emit('BOOLIT', trueTest)
-        pos += 4
-      } else {
-        emit('BOOLIT', falseTest)
-        pos += 5
-      }
+    // Literals
+    else if (atTrue) {
+      var lexeme = 'true'
+      emit('BOOLIT', lexeme)
+      pos += lexeme.length
+    }
+    else if (atFalse) {
+      var lexeme = 'false'
+      emit('BOOLIT', lexeme)
+      pos += lexeme.length
+    }
+    else if (atNull) {
+      var lexeme = 'null'
+      emit('NULLLIT', lexeme)
+      pos += lexeme.length
+    }
+    else if (atUndefined) {
+      var lexeme = 'undefined'
+      emit('UNDEFINEDLIT', lexeme)
+      pos += lexeme.length
```

Comment:
Yes, previously they did not have their type as the `kind`, see here in the parser where [we used to look for the lexeme, not the type](https://github.com/AuthorOfTheSurf/KobraScript/pull/58/files#diff-09461573a85e2d94f056dd6814769042R519) as in the surrounding code.

visual:
`{ kind: "UNDEFINEDLIT", lexeme: "undefined" }`
`{ kind: "undefined", lexeme: "undefined" }`